### PR TITLE
Python Install Env Vars: All CAPS

### PR DESCRIPTION
**Pull request by ax3l**
On Windows, environment variable names are case insensitive and often mapped to all CAPS spelling.

- https://stackoverflow.com/questions/7797269/python-environment-case-senstivity-os-environ
- https://en.wikipedia.org/wiki/Environment_variable#Syntax

This leads to an inconsistency between Unix and Windows like systems if we use case sensitive environment variable names for install control in Python.

Luckily, it is quite common to use all-CAPS variables in UNIX for environment controls as well, so we will make a breaking change to unify on that. Existing usage will see a warning of an unused variable with `python3 -m pip install -v .` (thanks to #2208)

### CODE REVIEW

**Review by EtherealMC-Bit — COMMENTED** [review]
#2208 Checks if there were unused environment variables that start with 'warpx', but does not care about capitalization. https://github.com/ECP-WarpX/WarpX/blob/a26a1ba1382a4ef2b3556197ddfa00f4dde6e6c1/setup.py#L208
So there won't be a warning associated with using an environment variable that isn't all caps from that portion of the code. An additional warning could be added to `setup.py` to cover the case that someone uses `WarpX` instead of `WARPX`.

**Comment by ax3l** [comment]
I think this will still warn:
- All exact spelling matches are `env.pop`ped (exact spelling as in Python code)
- That code matches everything that is left in `env`, using case insensitive comparison for anything that starts with `warpx`: key gets transformed to lowercase and then compared to `warpx` -> matches all spellings
- It prints the exact user-provided spelling

For instance, the exact code printed on Windows that the env variable `WARPX_COMPUTE` was found in `env` but was not `pop`ped:
https://github.com/ECP-WarpX/WarpX/blob/42b85398856eee72007f055fe0f6e2ae0d1ec75f/azure-pipelines.yml#L798

https://dev.azure.com/ECP-WarpX/WarpX/_build/results?buildId=4599
```
Building wheels for collected packages: pywarpx
  Running command git rev-parse HEAD
  c65d27b642a2b178870c92811feb77aa3dd8633c
  Running command 'C:\hostedtoolcache\windows\Python\3.7.9\x64\python.exe' 'C:\hostedtoolcache\windows\Python\3.7.9\x64\lib\site-packages\pip\_vendor\pep517\in_process\_in_process.py' build_wheel 'C:\Users\VSSADM~1\AppData\Local\Temp\tmp3px1zych'
  Building wheel for pywarpx (PEP 517): started

  WARNING: Found environment variable 'WARPX_COMPUTE', which is not a recognized WarpX option

  running bdist_wheel
  running build
...
```

**Review by EtherealMC-Bit — APPROVED** [review]
Ah in that case looks good to me.

File: setup.py [hunk 1]
```diff
@@ -177,32 +177,34 @@ def build_extension(self, ext):
...
-WarpX_EB = env.pop('WarpX_EB', 'OFF')
-WarpX_OPENPMD = env.pop('WarpX_OPENPMD', 'OFF')
-WarpX_PRECISION = env.pop('WarpX_PRECISION', 'DOUBLE')
-WarpX_PSATD = env.pop('WarpX_PSATD', 'OFF')
-WarpX_QED = env.pop('WarpX_QED', 'ON')
-WarpX_QED_TABLE_GEN = env.pop('WarpX_QED_TABLE_GEN', 'OFF')
-WarpX_DIMS = env.pop('WarpX_DIMS', '2;3;RZ')
+#   note: we use all-uppercase variable names for environment control to be
+#         consistent across platforms (especially Windows)
+WarpX_COMPUTE = env.pop('WARPX_COMPUTE', 'OMP')
```
Comment (dpgrote) [inline setup.py]: Perhaps make the Python variables also all caps as a reminder that they are from environment variables.

Comment (ax3l) [reply setup.py]: Ok, done.

**Review by dpgrote — APPROVED** [review]
Looks good!

**Pull request merged**